docs/developer-guide/etendo-classic/how-to-guides/how-to-create-a-navigation-bar-component.md

```diff
@@ -0,0 +1,171 @@
+  - Etendo Classic
+---
+
+#  How to create a Navigation Bar Component
+  
+##  Overview
+
+This section explains how a component can be added to the Etendo main navigation bar. Navigation bar components are shown in the top of the Etendo Classic layout. They are positioned from left to right.
+
+Some main features of the Etendo Classic navigation bar components:
+
+  * A navigation bar component can be any [Smartclient](https://smartclient.com/product/documentation.jsp){target="blank"} canvas. 
+  * Modules can provide new navigation bar components. 
+  * The position of a navigation bar component can be controlled. 
+  * Navigation bar components can be enabled by role. 
+
+  
+![](../../../assets/developer-guide/etendo-classic/how-to-guides/How_to_create_a_Navigation_Bar_Component-0.png)
+
+##  Example Module
+
+This is supported by an example module which shows examples of the code shown and discussed.
+
+  The code of this module can be downloaded from [this repository](https://github.com/etendosoftware/com.etendoerp.client.application.examples/tree/main/src/com/etendoerp/client/application/examples).
+
+
+##  Main Flow of the Navigation Bar Generation
+
+The navigation bar generation goes through a number of steps:
+
+  1. The user logs in, and navigates to the start page.
+  2. The start page builds the main layout consisting of the navigation bar and the main content area (with tabs).
+  3. The navigation bar is generated on the server as JavaScript (which is send to the browser). 
+  4. this is done by the main layout component. This component creates the overall JavaScript structure and then reads the navigation bar components from the navigation bar component table (using role information). 
+  5. Each navigation bar component is instantiated, its template is set and the generate method is called which generates the JavaScript of that component (using the template). 
+  6. The JavaScript of each component is assumed to create a single canvas or an array of Smartclient canvasses. The JavaScript (i.e. navigation bar component) is placed as a member of the horizontal layout, which builds the navigation bar. 
+
+This main flow illustrates that each navigation bar component can implement its own visualization by providing/using a custom template and component.
+
+##  Implementing a Navigation Bar Component
+
+To create a component which is shown in the navigation bar, the following parts need to be implemented:
+
+  * Create a Java class (the component) which represents the navigation bar component on the server 
+  * Create a template which generates the JavaScript which creates the component on the client 
+  * Register the navigation bar component in the navigation bar component table
+
+Each of these steps is described in more detail below.
+
+The example module contains a "Hello World" component with a template. This example adds a button to the navigation bar which (when clicked) will say hello to the current user.
+
+
```

Comment:
Instead of the image you should add a code block like this one 
https://github.com/etendosoftware/docs/blob/main/docs/developer-guide/etendo-copilot/how-to-create-copilot-tools.md

Reply:
Thanks for the suggestion, I will adapt it now